custom_components/unifiprotect/data.py

```diff
@@ -168,3 +172,12 @@ def async_signal_device_id_update(self, device_id: str) -> None:
 
         for update_callback in self._subscriptions[device_id]:
             update_callback()
+
+    @callback
+    def async_add_access_token_entity(self, entity: AccessTokenMixin) -> None:
+        # AccessTokenMixin is always an instance of UnifiProtectEntity
+        self._access_token_entities[entity.device.id] = entity  # type: ignore
+
+    @callback
+    def async_get_access_tokens_entity(self, entity_id: str) -> AccessTokenMixin | None:
+        return self._access_token_entities.get(entity_id)
```

Comment:
Storing entities in a shared container is not allowed.

Reply:
That is what `homeassistant.helpers.entity_component.EntityComponent`. Since we have our own data storage class, I was just kind of reimplementing that. Otherwise, I need to get the actually real entity class (not the `ReigstryEntity`) to get the access tokens.

Reply:
I think you can work around this by having an object to class the access tokens so the class objects end up in a shared container but the entities themselves don't

Reply:
Maybe should I instead store the access tokens on `protect_data` and then fetch them into the entity? Since it had a reference to protect data.

Reply:
I think that solves it as well